src/charm.py

```diff
@@ -72,11 +74,20 @@ def __init__(self, *args):
         """Initialize defaults and event handlers."""
         super().__init__(*args)
 
+        self._database = DatabaseObserver(self)
+
+        self.framework.observe(
+            self._database.database.on.database_created, self._database_relation_changed
+        )
+        self.framework.observe(
+            self._database.database.on.endpoints_changed, self._database_relation_changed
+        )
+        self.framework.observe(
```

Comment:
I believe that the database events are here for the same reason that I put them there on the first version of Synapse PR to add database integration, right? :)
Do you think is possible to use the same solution by passing pebble/container as a parameter?

Reply:
I propose to do this sort of changes as part of [ISD-934](https://warthogs.atlassian.net/browse/ISD-934). See https://github.com/canonical/discourse-k8s-operator/pull/79#discussion_r1274056536.